src/Randomizer.SMZ3.Tracking/VoiceCommands/CheatsModule.cs

```diff
@@ -33,30 +38,109 @@ public CheatsModule(Tracker tracker, ILogger<AutoTrackerModule> logger) : base(t
         {
             _logger = logger;
 
-            AddCommand("Heal player", HealPlayerRule(), (tracker, result) =>
+            AddCommand("Enable cheats", GetEnableCheatsRule(), (tracker, result) =>
+            {
+                _cheatsEnabled = true;
+                Tracker.Say(x => x.Cheats.EnabledCheats);
+            });
+
+            AddCommand("Disable cheats", GetDisableHintsRule(), (tracker, result) =>
             {
-                tracker.GameService?.TryHealPlayer();
+                _cheatsEnabled = false;
+                Tracker.Say(x => x.Cheats.DisabledCheats);
+            });
+
+            AddCommand("Fill rule", FillRule(), (tracker, result) =>
+            {
+                var fillType = result.Semantics.ContainsKey(s_fillCheatKey) ? (string)result.Semantics[s_fillCheatKey].Value : s_fillHealthChoices.First();
+                Fill(fillType);
             });
 
             AddCommand("Give item", GiveItemRule(), (tracker, result) =>
             {
                 var item = GetItemFromResult(tracker, result, out var itemName);
-                tracker.GameService?.TryGiveItem(item);
+                GiveItem(item);
             });
         }
 
-        private GrammarBuilder HealPlayerRule()
+        private bool PlayerCanCheat()
+        {
+            if (!_cheatsEnabled)
+            {
+                Tracker.Say(x => x.Cheats.PromptEnableCheats);
+                return false;
+            }
+            else if (Tracker.AutoTracker == null || !Tracker.AutoTracker.IsConnected)
+            {
+                Tracker.Say(x => x.Cheats.PromptEnableAutoTracker);
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Fills different types of pools for the player (health, bombs, missiles, etc.)
+        /// </summary>
+        /// <param name="fillType">What should be filled</param>
+        private void Fill(string fillType)
+        {
+            if (!PlayerCanCheat()) return;
+
+            if (s_fillHealthChoices.Contains(fillType) && Tracker.GameService?.TryHealPlayer() == true)
+            {
+                Tracker.Say(x => x.Cheats.CheatPerformed);
+            }
+            else
+            {
+                Tracker.Say(x => x.Cheats.CheatFailed);
+            }
+        }
+
+        private void GiveItem(ItemData? item)
+        {
+            if (!PlayerCanCheat()) return;
+
+            if (item != null && Tracker.GameService?.TryGiveItem(item) == true)
+            {
+                Tracker.Say(x => x.Cheats.CheatPerformed);
+            }
+            else
+            {
+                Tracker.Say(x => x.Cheats.CheatFailed);
+            }
+        }
+
+        private static GrammarBuilder GetEnableCheatsRule()
         {
+            return new GrammarBuilder()
+                .Append("Hey tracker, ")
+                .OneOf("enable", "turn on")
+                .OneOf("cheats", "cheat codes");
+        }
```

Comment:
It'd be real fun if we could accept some common cheat code enabling commands as references, e.g. "Hey tracker, turn on sv_cheats"